src/callbacks_stage/antidiffusive_stage.jl

```diff
@@ -0,0 +1,73 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+"""
+    AntidiffusiveStage()
+
+Perform antidiffusive stage for IDP limiting.
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct AntidiffusiveStage end
```

Comment:
What would happen if you tried to use this with, say, `SSPRK33` from OrdinaryDiffEq.jl?

Reply:
Right now, the problem mentioned in [this comment](https://github.com/trixi-framework/Trixi.jl/pull/1476#discussion_r1215356222) occurs. This means that the needed containers are not yet resized. We have thought about moving the resize call, but haven't found a good place yet.

But even if this issue would be fixed and sometimes the simulation could run through for some problems, the limiting would not work as desired, because it would take place at the wrong place of the RK stage. (The IDP limiting is only designed for the Forward Euler step. That's why we use SSPRK methods, which can be represented as such.)

Reply:
Would it be possible to allow general ODE.jl integrators by using the antidiffusive stage as both a stage callback *and* a step callback? Or will this not help either?

Reply:
Do you mean inserting something like `integrator.alg::SimpleAlgorithmSSP` into one of the functions?

Reply:
Yes. Maybe around `function (antidiffusive_stage!::AntidiffusiveStage)(u_ode, integrator, stage)` or so

Reply:
That sounds good. Right now, the callbacks are precompiled before the time integration files. Therefore, the algorithms and the SSP integrator are not defined there. Would it be ok to just change the order of precompilation?

Reply:
If that works and is required - yes, sure